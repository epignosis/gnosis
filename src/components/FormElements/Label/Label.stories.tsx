import React from "react";
import { Story } from "@storybook/react";
import LabelComponent, { LabelProps } from "./Label";

export default {
  title: "components/Form Elements/Label",
  component: LabelComponent,
  argTypes: {
    size: {
      control: {
        type: "select",
        options: ["md", "lg"],
      },
    },
  },
};

export const Label: Story<LabelProps> = (args) => (
  <LabelComponent {...args}>This is an input label</LabelComponent>
);

Label.args = {
  size: "md",
};
