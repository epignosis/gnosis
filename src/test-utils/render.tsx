import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
