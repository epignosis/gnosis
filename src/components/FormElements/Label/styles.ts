import { css, Theme, SerializedStyles } from "@emotion/react";
import { LabelSize } from "./Label";

export const label = (
  { typeScaleSizes }: Theme,
  { size }: { size: LabelSize },
): SerializedStyles => {
  const fontSizes = {
    md: typeScaleSizes.sm,
    lg: typeScaleSizes.md,
  };

  return css`
    font-size: ${fontSizes[size]};
    font-weight: 700;
    display: block;
    margin: 0 0 0.5rem;
    white-space: nowrap;
  `;
};
