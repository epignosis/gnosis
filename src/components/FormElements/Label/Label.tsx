import React, { FC } from "react";
import { label } from "./styles";

export type LabelSize = "md" | "lg";

export type LabelProps = React.LabelHTMLAttributes<HTMLLabelElement> & {
  size?: LabelSize;
};

const Label: FC<LabelProps> = ({ size = "md", children, ...rest }) => {
  return (
    <label css={(theme) => label(theme, { size })} {...rest}>
      {children}
    </label>
  );
};

export default Label;
