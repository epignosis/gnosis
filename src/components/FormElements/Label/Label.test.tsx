import React from "react";
import faker from "faker";
import Label from "./Label";
import { render, screen } from "@test-utils/render";

describe("<Label />", () => {
  it("renders correctly", () => {
    const labelTxt = faker.random.alpha();
    render(<Label>{labelTxt}</Label>);

    const label = screen.getByText(labelTxt);

    expect(label).toHaveTextContent(labelTxt);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Label id="my-label" className="horizontal-label">
        Test label
      </Label>,
    );

    expect(container).toMatchSnapshot();
  });
});
