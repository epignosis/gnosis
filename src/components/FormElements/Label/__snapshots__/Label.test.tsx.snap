// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> matches snapshot 1`] = `
<div>
  <label
    class="css-xs6o3g-label-label"
  >
    Test label
  </label>
</div>
`;
